### ISSUE TITLE
potential bug 2

### ISSUE THREAD

**Issue by dwaithe** [WeidiXie/cell_counting_v2#2]
Hi Weidi,

I noticed your learning decay function decay function is probably bugged. After 48 iterations it will go up to 1e-3 again, giving you a fluctuating learning-rate, if you have more than 48 epochs. This is bad right? This isn't just a special learning regime?  

![image](https://cloud.githubusercontent.com/assets/4719022/23255711/86217c36-f9b4-11e6-9c60-fccc6928eac0.png)

Present function:
```
def step_decay(epoch):
    step = 16
    num =  epoch // step 
    if num % 3 == 0:
        lrate = 1e-3
    elif num % 3 == 1:
        lrate = 1e-4
    else:
        lrate = 1e-5
        #lrate = initial_lrate * 1/(1 + decay * (epoch - num * step))
    print('Learning rate for epoch {} is {}.'.format(epoch+1, lrate))
    return np.float(lrate)
```

The following works for me:
```
def step_decay(epoch):
    step = 16
    num =  epoch // step
    if num >1:
        lrate = 1e-5
    elif num % 3 == 0:
        lrate = 1e-3
    elif num % 3 == 1:
        lrate = 1e-4
    
        #lrate = initial_lrate * 1/(1 + decay * (epoch - num * step))
    print('Learning rate for epoch {} is {}.'.format(epoch+1, lrate))
    return np.float(lrate)
```

**Comment by WeidiXie** [comment]
Hi, thanks for pointing out this.

As we discussed in the E-mail, this is not a bug, just an intuitive way to escape from the saddle points. I would recommend the readers to several related papers,
[1]: SGDR: Stochastic Gradient Descent with Warm Restarts
[2]: Snapshot Ensembles: Train 1, Get M for Free

In this code, I prefer simplicity, so I didn't use any sophisticated learning rate schedule, you can try the cyclic learning rate or others.
